pkg/skaffold/runner/runner.go

```diff
@@ -261,6 +262,11 @@ func (r *SkaffoldRunner) buildTestDeploy(ctx context.Context, out io.Writer, art
 	// Make sure all artifacts are redeployed. Not only those that were just built.
 	r.builds = build.MergeWithPreviousBuilds(bRes, r.builds)
 
+	if r.runCtx.Opts.ManualDeploy && r.HasDeployed() {
```

Comment:
Can we just skip deploy here and return? This assumes the flow is
buildTrigger -> deployTrigger -> buildTrigger -> deployTrigger

But users might do:
buildTrigger -> buildTrigger -> deployTrigger

Reply:
this is part of `buildTestDeploy()` so we need to do the deploy here. the reason this is checking `r.HasDeployed()` is because this is the entrypoint for `dev`, and we want to do the first build and deploy before accepting manual triggers.

you're right though, build -> build -> deploy is a flow that users might want to do and it's not going to work exactly that way in this function. originally I decided it was enough to let the triggers get "queued" and then execute when that part of the dev loop came up, but now I'm rethinking it. gonna take a stab at supporting consecutive builds without a deploy.

Reply:
cool!